app/api/resources/metadata.py

```diff
@@ -212,15 +213,20 @@ async def get_genome_ftplinks(request: Request, genome_uuid: str, region_name: s
     "/genome/{genome_uuid}/dataset/{dataset_type}/attributes", name="dataset_attributes"
 )
 async def get_genome_dataset_attributes(
-    request: Request, genome_uuid: str, dataset_type: str
+    request: Request,
+    genome_uuid: str,
+    dataset_type: str,
+    attribute_name: Annotated[list[str] | None, Query()] = None,
 ):
     try:
         dataset_attributes = MessageToDict(
             grpc_client.get_dataset_attributes(
-                genome_uuid=genome_uuid, dataset_type=dataset_type
+                genome_uuid=genome_uuid,
+                dataset_type=dataset_type,
+                attribute_name=attribute_name,
             )
         )
-        if dataset_attributes == {}:
+        if len(dataset_attributes) <= 1:
```

Comment:
What does it mean for `len(dataset_attributes)` to be 1? Why is it also wrong?

Reply:
For some reason when you filter GRPC request based on attribute name you will get a dictionary containing "release_version": 110.1

Reply:
I've just checked using Production's grpcui:

1) Request `genebuild` dataset for `a7335667-93e7-11ec-a39d-005056b38ce3` genome, and only ask for `genebuild.average_cds_length` attribute:

```
{
  "attributes": [
    {
      "attribute_name": "genebuild.average_cds_length",
      "attribute_value": "1191.97",
      "dataset_version": "GENCODE44",
      "dataset_uuid": "949defef-c4d2-4ab1-8a73-f41d2b3c7719",
      "dataset_type": "genebuild"
    }
  ],
  "release_version": 110.1
}
```

2) Same, but ask for a non-existent attribute `blah`:

```
{
  "attributes": [],
  "release_version": 110.1
}
```

3) Same, but ask for a non-existent dataset `foo`:

```
{
  "attributes": [],
  "release_version": 0
}
```

Are you getting different payloads? If they are the same payloads, doesn't this mean we should be checking not the length of the `dataset_attributes` dictionary, but rather the length of the `dataset_attributes["attributes"]` array? `release_version` is kind of irrelevant, no?

Reply:
This is what I get with an non existing keyword:
{'releaseVersion': 110.1}

And it doesn't make a difference either way of checking if the data exists.

Reply:
Also this is not the only case for 404, if genome id doesn't exists or dataset name,...

Reply:
I agree that it doesn't make a difference; but I bet `if len(dataset_attributes) <= 1:` will be puzzling to the reader — they will probably wonder why `1` is a special case. Shouldn't the check be more explicit, like `if dataset_attributes.get('attributes') && len(dataset_attributes.get('attributes')) == 0`? Or perhaps the `DatasetAttributes` class should have an `is_empty` method? Or at least there should be a comment explaining why `len(dataset_attributes) <= 1` is bad 🙂

Reply:
It's not puzzling since it's clearly a check for 404 status, but I updated the code so it wouldn't be puzzling for you.